chainer/utils/type_check.py

```diff
@@ -504,26 +504,103 @@ def __init__(self, expect, actual, msg=None):
+for op in _bool_ops:
+    setattr(_MissingArgument, op, lambda self, other: self)
 
-    This function also asserts that lenghts of in_types and names are the
-    same.
+# Set binary operators to _MissingArgument
+_bin_ops = ['__add__', '__radd__', '__sub__', '__rsub__', '__mul__',
+            '__rmul__']
+if sys.version_info < (3, 0, 0):
+    _bin_ops += ['__div__', '__rdiv__']
+else:
+    _bin_ops += ['__truediv__', '__rtruediv__']
+_bin_ops += ['__floordiv__', '__rfloordiv__', '__mod__', '__rmod__', '__pow__',
+             '__lshift__', '__rlshift__', '__rshift__', '__rrshift__',
+             '__and__', '__rand__', '__xor__', '__rxor__', '__or__', '__ror__']
+for op in _bin_ops:
+    setattr(_MissingArgument, op, lambda self, other: self)
+
+# Set unary operators to _MissingArgument
+_un_ops = ['__neg__', '__pos__', '__invert__']
+for op in _un_ops:
+    setattr(_MissingArgument, op, lambda self: self)
+
+_missing_argument = _MissingArgument()
+
+
+def argname(in_types, names, optional_names=()):
+    """Destructures the input types with assigning them user friendly names
+
+    This function also asserts that in_types has the appropriate length with
+    that of names.
 
     Args:
-        in_types (tuple of TypeInfoTuple): Tuple of type information to assign
-            name to.
-        names (tuple of str): Human-readabel names of ``in_types``.
+        in_types (TypeInfoTuple): Tuple of type information to destructure.
+        names (tuple of str): Human-readable names of required arguments in
+            ``in_types``.
+        optional_names (tuple of str): Human-readable names of optional
+            arguments in ``in_types``.
     """
-    if len(in_types) != len(names):
+    min_len = len(names)
+    max_len = min_len + len(optional_names)
+    actual_len = len(in_types)
+
+    if not min_len <= actual_len <= max_len:
+        if len(optional_names) == 0:
+            expected = '{} argument(s)'.format(len(names))
+        else:
+            expected = '{}-{} argument(s)'.format(
+                len(names), len(names) + len(optional_names))
         raise InvalidType(
-            '{} argument(s)'.format(str(len(names))),
-            '{} argument(s)'.format(str(len(in_types))),
+            expected, '{} argument(s)'.format(actual_len),
             'Invalid number of arguments')
+
+    ret = in_types
+
+    # Assign the names for the mandatory arguments.
+    # Note: zip truncates to the shorter (names) if the lengths are different
     for in_type, name in zip(in_types, names):
         if isinstance(in_type, Variable):
             in_type.name = name
 
+    # Assign the names for the optional arguments.
+    if actual_len > min_len:
+        for in_type, name in zip(in_types[len(names):], optional_names):
+            if isinstance(in_type, Variable):
+                in_type.name = name
+
+    # For missing optional arguments, fill with _missing_argument which
```

Comment:
This function is called many times. So, This code is only for test.
Can you remove it?